### PR TITLE
feat(LayoutEditor) Check field usage before delete

### PR DESCRIPTION
**Pull request by denaldd**

### CODE REVIEW

File: Smarty/templates/Settings/LayoutBlockEntries.tpl [hunk 1]
```diff
@@ -14,6 +14,7 @@
 <input type="hidden" name="mode">
 <script type="text/javascript" src="include/js/customview.js"></script>
 <script type="text/javascript" src="include/js/general.js"></script>
+        <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
```
Comment (joebordes) [inline Smarty/templates/Settings/LayoutBlockEntries.tpl]: Please use the jquery library we have in the application:

https://github.com/tsolucio/corebos/tree/master/include/jquery

File: Smarty/templates/Settings/LayoutBlockEntries.tpl [hunk 2]
```diff
@@ -387,14 +388,80 @@
...
+											<input type="button" name="delete" value="{$value.label}" id="empid" class="crmButton small delete"  onclick="getData(this.value, '{$value.label}'); fnvshobj(this,'hiddenfield_{$value.label}'); " />
 										{/if}
 										<input type="button" name="cancel" value=" {$APP.LBL_CANCEL_BUTTON_LABEL} " class="crmButton small cancel" onclick="fninvsh('editfield_{$value.fieldselect}');" />
 									</td>
 								</tr>
 							</table>
 						</div>
 					</div>
+
+<script type="text/javascript" src="modules/Settings/script.js"></script>
```
Comment (joebordes) [inline Smarty/templates/Settings/LayoutBlockEntries.tpl]: Please put this javascript code in the Settings.js file

https://github.com/tsolucio/corebos/blob/master/modules/Settings/Settings.js

that one is already included in the service worker

File: Smarty/templates/Settings/LayoutBlockEntries.tpl [hunk 3]
```diff
@@ -387,14 +388,80 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (joebordes) [inline Smarty/templates/Settings/LayoutBlockEntries.tpl]: Please eliminate all the empty lines if possible

File: modules/Settings/loaddata.php [hunk 4]
```diff
@@ -0,0 +1,174 @@
+<?php
+$Vtiger_Utils_Log = true;
```
Comment (joebordes) [inline modules/Settings/loaddata.php]: put license here:

https://github.com/tsolucio/corebos/blob/master/build/code_license.tpl#L50

File: modules/Settings/loaddata.php [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+include_once('vtlib/Vtiger/Module.php');
+global $current_user,$adb;
+set_time_limit(0);
+ini_set('memory_limit','1024M');
+$fldname = $_GET['empid'];
```
Comment (joebordes) [inline modules/Settings/loaddata.php]: filter this input with  vtlib_purify()

File: modules/Settings/loaddata.php [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+?>
+
+
+<?php
+
+// Workflow Conditions
+       $crs = $adb->query("SELECT workflow_id,summary
+                FROM `com_vtiger_workflows`
+                WHERE test like '%$fldname%'");
+            if ($crs and $adb->num_rows($crs)>0) {
```
Comment (joebordes) [inline modules/Settings/loaddata.php]: use pquery here to avoid sql injection

```
$crs = $adb->pquery('SELECT workflow_id,summary FROM `com_vtiger_workflows` WHERE test like ?', array('%'.$fldname.'%');
```

File: modules/Settings/script.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+function getData(empid, divid){
```
Comment (joebordes) [inline modules/Settings/script.js]: move this code to Settings.js

**Comment by joebordes** [comment]
Hi

Thank you for your contribution!  Please attend the inline comments I made and also read this article on code formatting. I would really appreciate you applied that code standard if possible.

http://corebos.org/documentation/doku.php?id=en:devel:developmentguidelines

**Comment by joebordes** [comment]
I merged and cleaned up the files (https://github.com/tsolucio/corebos/commit/72862c35b11d918748da34c9ce6b865732f1ff3e)

I also kept the "Delete" label as I think it is better.

There are two errors and two things we have to change. I will create issues and assign them to you. If you can't do them let me know.

Thanks!!

**Comment by joebordes** [comment]
the issues are: #237 #238 #239 #240 

let me know if you have any problems
Have a nice weekend

**Comment by denaldd** [comment]
I made the changes 8d646f53f56e459b3e71e9cdfc416cca98cbadda

**Comment by joebordes** [comment]
Issues #237 and #240 are still missing.

Thanks!
